src/plugins/irc-events/topic.js

```diff
@@ -39,7 +39,7 @@ module.exports = function(irc, network) {
 			type: Msg.Type.TOPIC_SET_BY,
 			mode: chan.getMode(data.nick),
 			nick: data.nick,
-			when: data.when,
+			when: new Date(data.when * 1000),
```

Comment:
@astorije Yes, because the Date object doesn't persist through server->client communication. Json doesn't have a Date type so it gets turned into a `2016-01-02T12:34:56T` format, which needs to be parsed back into a real date object.